### DISCUSSION TITLE
Hide a task

### DISCUSSION THREAD

**Discussion by LeMerP** [RexOps/Rex#1614] [Questions and answers] [answered]
Hello everyone,

is there a way to hide a task from getting listet by "rex -T" command?
I already tried
`task "test, hidden => 1, sub {};`
but that did not do the job.

I saw the "hidden" parameter in the Rex::Task module, but do not know how to pass this to my test task.

Kind regards,
Patrick

**Comment by ferki** [comment, marked as answer]
Hi @LeMerP,

there's an internal-facing `dont_register` option you may attach to task definitions, like this:

https://github.com/RexOps/Rex/blob/0f87255268b3d3e593bd7bf816dd801f42464ee1/lib/Rex/Commands/Box.pm#L418-L420

See also:

- [Basic anatomy of Rex task definitions](https://blog.ferki.it/2023/08/11/local-management-with-rex/#basicanatomyofrextaskdefinitions) for a quick summary on the general structure of task definitions
- [`include`](https://metacpan.org/pod/Rex::Commands#include-Module::Name) command to "Include a module without registering its tasks" (instead of loading modules with Perl's own `use` and `require`)

Happy hacking!
